util/progress/progressui/display.go

```diff
@@ -42,6 +43,13 @@ func WithDesc(text string, console string) DisplaySolveStatusOpt {
 	}
 }
 
+// WithJSONPrinter prints each line of output as a JSON stream instead of as raw text.
+func WithJSONPrinter() DisplaySolveStatusOpt {
```

Comment:
This should just be a `mode` parameter in `DisplaySolveStatus` so the output doesn't only depend on passing the console.

Reply:
But it is not very nice library if there are 3 distinct modes then to configure 2 of them you would use the console object and for the third there is a special method. And we will probably add even more modes in the future.

Buildctl also supports these progress modes so it is not buildx-only concept. We can just combine all the usage so it is defined in one place (basically push it in a little bit lower level).